init.zsh

```diff
@@ -3,49 +3,49 @@
 #
 
 autoload -Uz is-at-least
-if ! is-at-least 5.2; then
-  print "WARNING: Support for zsh < 5.2 will be removed in a future update. Update your version of zsh now to prevent any interruptions." >&2
-fi
-
-# Define zim location
-ZIM="${ZDOTDIR:-${HOME}}/.zim"
-
-# Source user configuration
-if [[ -s "${ZDOTDIR:-${HOME}}/.zimrc" ]]; then
-  source "${ZDOTDIR:-${HOME}}/.zimrc"
-fi
-
-load_zim_module() {
-  local wanted_module
-
-  for wanted_module (${zmodules}); do
-    if [[ -s "${ZIM}/modules/${wanted_module}/init.zsh" ]]; then
-      source "${ZIM}/modules/${wanted_module}/init.zsh"
-    elif [[ ! -d "${ZIM}/modules/${wanted_module}" ]]; then
-      print "No such module \"${wanted_module}\"." >&2
-    fi
-  done
-}
+if is-at-least 5.2; then
```

Comment:
rather than wrap this whole thing in a conditional, invert the test here and exit early.